ts/components/settings/section/CategoryPrivacy.tsx

```diff
@@ -12,19 +12,23 @@ import { TypingBubble } from '../../conversation/TypingBubble';
 import { SessionSettingButtonItem, SessionToggleWithDescription } from '../SessionSettingListItem';
 import { displayPasswordModal } from '../SessionSettings';
 
-async function toggleLinkPreviews() {
-  const newValue = !window.getSettingValue(SettingsKey.settingsLinkPreview);
-  await window.setSettingValue(SettingsKey.settingsLinkPreview, newValue);
-  if (!newValue) {
-    await Data.createOrUpdateItem({ id: hasLinkPreviewPopupBeenDisplayed, value: false });
-  } else {
+async function toggleLinkPreviews(isToggleOn: boolean, setIsToggleOn: (value: boolean) => void) {
+  if (!isToggleOn) {
```

Comment:
would be nicer to check for the value stored in DB here too, so rather than having the argument passed
have another 
```ts
const isLinkPreviewsOn = Boolean(window.getSettingValue(SettingsKey.settingsLinkPreview))
```

Reply:
Also, pass the forceUpdate function to this `toggleLinkPreviews` function so rather than `setIsToggleOn;` you can just call forceUpdate()